test/integration/daemonset/daemonset_test.go

```diff
@@ -498,6 +533,50 @@ func TestSimpleDaemonSetLaunchesPods(t *testing.T) {
 	})
 }
 
+func TestSimpleDaemonSetRestartsPods(t *testing.T) {
```

Comment:
it would be nice to have an e2e that sets the pod phase to succeeded and verify the daemonset controller recreates it.

Reply:
this is already an integration test. I don't think an E2E environment would change much.

Reply:
+1 I think the integration test already covers that indeed. 

There might be value added of e2e test, but when we perform an actual graceful node shutdown, rather than just setting the phase to Succeeded. WDYT?